keepassxc-browser/background/keepass.js

```diff
@@ -1062,3 +1048,23 @@ keepass.updatePopup = function(iconType) {
         browserAction.show(null, {'id': page.currentTabId});
     }
 };
+
+// Updates the database hashes to content script
+keepass.updateDatabase = function() {
+    keepass.testAssociation((associationResponse) => {
+        keepass.isConfigured().then((configured) => {
+            keepass.updatePopup(configured ? 'normal' : 'cross');
+
+            // Send message to content script
+            browser.tabs.query({ active: true, currentWindow: true }).then((tabs) => {
+                if (tabs.length) {
+                    browser.tabs.sendMessage(tabs[0].id, {
+                        action: 'check_database_hash',
+                        hash: {old: keepass.previousDatabaseHash, new: keepass.databaseHash}
+                    });
+                    keepass.previousDatabaseHash = keepass.databaseHash;
```

Comment:
If we are setting the previous hash to the "new" hash, when is the "new" hash set? This function makes no sense to me.

Reply:
The new hash is set when getDatabaseHash() has a response. This is called quite often.

Reply:
Wouldn't this be subject to race conditions then?

Reply:
No. Updating these values cannot happen at the same time. In this case the update function is called only when user is locking or unlocking the database from KeePassXC or from the extension popup. And updating the current database hash only happens when doing some operation (retrieving credentials at page load etc).